sov-modules/sov-app-template/src/lib.rs

```diff
@@ -136,9 +132,9 @@ where
             // Run the stateful verification, possibly modifies the state.
             let verified_tx = match self.tx_hooks.pre_dispatch_tx_hook(tx, &mut batch_workspace) {
                 Ok(verified_tx) => verified_tx,
-                Err(_) => {
+                Err(e) => {
                     self.revert_and_slash(batch_workspace);
-                    return Err(ConsensusSetUpdate::slashing(sequencer));
+                    anyhow::bail!("Stateful verification error: {}", e);
```

Comment:
So the exact reason is inside `e`. Now it can be only bad nonce but in the future it might be not enough balance to cover fees etc (but the correct message will be in e )